src/views/PurchaseOrderDetail.vue

```diff
@@ -151,7 +150,7 @@ export default defineComponent({
       this.store.dispatch("purchaseOrder/getOrderDetails", {payload, orderId});
     }
   }, 
-  mounted(){
+  mounted() {
     this.getorderDetails(this.$route.params.slug);
```

Comment:
Here, we have defined a method which is called on mounted and passing route param here. Now when a user clicks on a PO, then the API will only be called once the PO detail view is mounted, so instead, I think you should call this API on the click event of the PO page and not on the mounted hook of details page.

Reply:
Okay, Sir, I will implement it.